openrtb_ext/request_wrapper.go

```diff
@@ -837,6 +927,26 @@ func (de *DeviceExt) SetPrebid(prebid *ExtDevicePrebid) {
 	de.prebidDirty = true
 }
 
+func (de *DeviceExt) Clone() *DeviceExt {
+	if de == nil {
+		return nil
+	}
+
+	clone := *de
+	clone.ext = maputil.Clone(de.ext)
+
+	if de.prebid != nil {
+		newPrebid := *de.prebid
+		if newPrebid.Interstitial != nil {
+			newInterstitial := *de.prebid.Interstitial
```

Comment:
`DeviceExt.Prebid.Interstitial` is a pointer to `ExtDeviceInt` so you'd need to create a copy of that in a separate `Interstitial` object and then assign that to `newPrebid.Interstitial`

Reply:
Thanks, good catch

Reply:
Actually, that is the `newInterstitial` object up there. I went in to figure out why my tests didn't catch this, and discovered that all pointers are being cloned properly. These things are a mess of pointers certainly.